### PR TITLE
IDENTITY-6299: User creation via SCIM API does not return updated att…

### PR DESCRIPTION
**Pull request by lmanohara**
…ribute values of the user

### CODE REVIEW

File: components/org.wso2.carbon.identity.scim.provider/src/main/java/org/wso2/carbon/identity/scim/provider/impl/SCIMUserManager.java [hunk 1]
```diff
@@ -169,8 +169,44 @@ public User createUser(User user, boolean isBulkUserAdd) throws CharonException,
                 if (claimsMap.containsKey(SCIMConstants.USER_NAME_URI)) {
                     claimsMap.remove(SCIMConstants.USER_NAME_URI);
                 }
-                carbonUM.addUser(user.getUserName(), user.getPassword(), null, claimsMap, null);
+
+                if (claimsMap.containsKey(SCIMConstants.META_LOCATION_URI)) {
```
Comment (darshanasbg) [inline components/org.wso2.carbon.identity.scim.provider/src/main/java/org/wso2/carbon/identity/scim/provider/impl/SCIMUserManager.java]: Add a comment mentioning the reason to remove location claim

File: components/org.wso2.carbon.identity.scim.provider/src/main/java/org/wso2/carbon/identity/scim/provider/impl/SCIMUserManager.java [hunk 2]
```diff
@@ -169,8 +169,44 @@ public User createUser(User user, boolean isBulkUserAdd) throws CharonException,
...
+                    claimsMap.remove(SCIMConstants.META_LOCATION_URI);
+                }
+
+                Map<String, String> clonedClaimsMap = new HashMap(claimsMap);
+                carbonUM.addUser(user.getUserName(), user.getPassword(), null, clonedClaimsMap, null);
+                //Remove username from the modified map before compare with the original map
```
Comment (darshanasbg) [inline components/org.wso2.carbon.identity.scim.provider/src/main/java/org/wso2/carbon/identity/scim/provider/impl/SCIMUserManager.java]: Add the **reason** to remove modified username.

Comment (darshanasbg) [reply components/org.wso2.carbon.identity.scim.provider/src/main/java/org/wso2/carbon/identity/scim/provider/impl/SCIMUserManager.java]: Add a new line before the comment.

And add a space before starting the text. ie.
`// Remove`

File: components/org.wso2.carbon.identity.scim.provider/src/main/java/org/wso2/carbon/identity/scim/provider/impl/SCIMUserManager.java [hunk 3]
```diff
@@ -169,8 +169,44 @@ public User createUser(User user, boolean isBulkUserAdd) throws CharonException,
...
+                String modifiedUserName = null;
+                if (clonedClaimsMap.containsKey(SCIMConstants.USER_NAME_URI)) {
+                    modifiedUserName = clonedClaimsMap.get(SCIMConstants.USER_NAME_URI);
+                    clonedClaimsMap.remove(SCIMConstants.USER_NAME_URI);
+                }
                 log.info("User: " + user.getUserName() + " is created through SCIM.");
```
Comment (darshanasbg) [inline components/org.wso2.carbon.identity.scim.provider/src/main/java/org/wso2/carbon/identity/scim/provider/impl/SCIMUserManager.java]: Move the info log to the end of the method

File: components/org.wso2.carbon.identity.scim.provider/src/main/java/org/wso2/carbon/identity/scim/provider/impl/SCIMUserManager.java [hunk 4]
```diff
@@ -169,8 +169,44 @@ public User createUser(User user, boolean isBulkUserAdd) throws CharonException,
...
+
+                // Check if the user claims map passed has been modified.
+                if (!claimsMap.keySet().containsAll(clonedClaimsMap.keySet()) || !clonedClaimsMap.keySet()
```
Comment (darshanasbg) [inline components/org.wso2.carbon.identity.scim.provider/src/main/java/org/wso2/carbon/identity/scim/provider/impl/SCIMUserManager.java]: use,
`claimsMap.equals(clonedClaimsMap)`

File: components/org.wso2.carbon.identity.scim.provider/src/main/java/org/wso2/carbon/identity/scim/provider/impl/SCIMUserManager.java [hunk 5]
```diff
@@ -169,8 +169,44 @@ public User createUser(User user, boolean isBulkUserAdd) throws CharonException,
...
+                    String userId = clonedClaimsMap.get(SCIMConstants.ID_URI);
+                    clonedClaimsMap.put(SCIMConstants.META_LOCATION_URI, SCIMCommonUtils.getSCIMUserURL(userId));
+
+                    //construct the SCIM Object from the attributes
+                    try {
+                        User newUser = (User) AttributeMapper
+                                .constructSCIMObjectFromAttributes(clonedClaimsMap, SCIMConstants.USER_INT);
+                        return newUser;
+                    } catch (NotFoundException e) {
+                        log.warn("Failed to populate modified claims for user : " + user.getUserName() + " created.",
```
Comment (darshanasbg) [inline components/org.wso2.carbon.identity.scim.provider/src/main/java/org/wso2/carbon/identity/scim/provider/impl/SCIMUserManager.java]: throw an exception, rather logging

Comment (lmanohara) [reply components/org.wso2.carbon.identity.scim.provider/src/main/java/org/wso2/carbon/identity/scim/provider/impl/SCIMUserManager.java]: This is an API change and this should specifically handle in the charon level.

**Pull request merged**